hack/istio/multicluster/kiali-prepare-remote-cluster.sh

```diff
@@ -245,6 +245,13 @@ create_kiali_remote_cluster_secret() {
     local cert_auth_yaml="certificate-authority-data: ${remote_cluster_ca_bytes}"
   fi
 
+  # Check if the REMOTE_CLUSTER_NAME matches the AWS EKS ARN regex pattern.
+  # If it does, extract the cluster name from the ARN and assign it back to REMOTE_CLUSTER_NAME.
+  AWS_REGION_REGEX='(us|eu|ap|sa|ca|me)-(north|south|east|west|central)-[0-9]'
```

Comment:
This code now works. However, I did some reading of the Amazon docs and I believe you need to enhance the regex here.

First, I do want to document here that Amazon account IDs are really 12-digit numbers. I was curious if account IDs could have hex-characters (e.g. `a` through `f`) and if they could be different than 12 digits long - but none of that is the case. See Amazon doc [here](https://docs.aws.amazon.com/accounts/latest/reference/manage-acct-identifiers.html). So the regex is correct; as per docs:
```
AWS account ID
A 12-digit number, such as 012345678901, that uniquely identifies an AWS account.
```

Now follow the link found in that Amazon doc page to [Amazon Resource Names (ARNs)](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) where the ARN format is specified. I see under `region` this:
```
region
The Region code. For example, us-east-2 for US East (Ohio). For the list of Region codes, see Regional endpoints in the AWS General Reference.
```

Follow the link to [Regional Endpoints](https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints) to get the list of region codes.

The codes are more than what this PR's regex has. This PR is missing Africa (`af`), Israel (`il`), and US Government (`us-gov`). This PR is also missing `southeast` and `northeast` (as a side note for the curious, there is currently no "southwest" or "northwest" regions supported by Amazon).

So the region regex could be this:
```
AWS_REGION_REGEX='(us|eu|ap|sa|ca|me|af|il|us-gov)-(north|south|east|west|central|southeast|northeast)-[0-9]'
```

However, even that isn't right because there is no `af-east-9` for example (where this regex would allow it). In fact, there are no regions with a index number higher than 4 so that `[0-9]` could be `[0-4]`. 

To be technically correct, we'd have to list each of the 31 regions in the regex like this:

```
AWS_REGION_REGEX='(us-east-2|us-east-1|us-west-1|us-west-2|af-south-1|ap-east-1|ap-south-2|ap-southeast-3|ap-southeast-4|ap-south-1|ap-northeast-3|ap-northeast-2|ap-southeast-1|ap-southeast-2|ap-northeast-1|ca-central-1|ca-west-1|eu-central-1|eu-west-1|eu-west-2|eu-south-1|eu-west-3|eu-south-2|eu-north-1|eu-central-2|il-central-1|me-south-1|me-central-1|sa-east-1|us-gov-east-1|us-gov-west-1)'
```

I don't know how often Amazon adds or removes supported regions, so the first form of the regex might be better for future additions (e.g. perhaps a third us west region may be added `us-west-3`). So I'm inclined to not worry about things like `af-southeast-9` and just support the original regex with the addition of `af`, `il`, `us-gov`, `southeast`, and `northeast`:

```suggestion
  AWS_REGION_REGEX='(us|eu|ap|sa|ca|me|af|il|us-gov)-(north|south|east|west|central|southeast|northeast)-[0-9]'
```